staging/src/k8s.io/apiextensions-apiserver/test/integration/testserver/resources.go

```diff
@@ -160,20 +211,33 @@ func CreateNewCustomResourceDefinitionWatchUnsafe(crd *apiextensionsv1beta1.Cust
 		return err
 	}
 
-	// wait until the resource appears in discovery
-	err = wait.PollImmediate(500*time.Millisecond, 30*time.Second, func() (bool, error) {
-		resourceList, err := apiExtensionsClient.Discovery().ServerResourcesForGroupVersion(crd.Spec.Group + "/" + crd.Spec.Version)
-		if err != nil {
-			return false, nil
-		}
-		for _, resource := range resourceList.APIResources {
-			if resource.Name == crd.Spec.Names.Plural {
-				return true, nil
+	versions := []string{}
+
+	if len(crd.Spec.Versions) != 0 {
+		for _, v := range crd.Spec.Versions {
+			if v.Served {
+				versions = append(versions, v.Name)
 			}
 		}
-		return false, nil
-	})
+	} else {
+		versions = append(versions, crd.Spec.Version)
+	}
 
+	for _, version := range versions {
```

Comment:
one version is enough, let's use the first. Our discovery registration code returns all or none.